client/my-sites/earn/components/add-edit-coupon-modal/test/index.jsx

```diff
@@ -0,0 +1,280 @@
+	} );
+
+	test( 'should show discount amount input when discount type is amount', async () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal closeDialog={ closeDialog } coupon={ {} } />
+		);
+		expect( screen.getByRole( 'option', { name: 'Amount' } ).selected ).toBe( false );
+		expect( screen.getByRole( 'option', { name: 'Percentage' } ).selected ).toBe( true );
+		expect( screen.getByLabelText( 'Amount' ) ).toMatchSnapshot();
+		await userEvent.selectOptions(
+			screen.getByLabelText( 'Discount type' ),
+			screen.getByRole( 'option', { name: 'Amount' } )
+		);
+
+		expect( screen.getByRole( 'option', { name: 'Amount' } ).selected ).toBe( true );
+		expect( screen.getByRole( 'option', { name: 'Percentage' } ).selected ).toBe( false );
+		expect( screen.getByLabelText( 'Amount' ) ).toMatchSnapshot();
+	} );
+
+	test( 'should enable select dropdown if duration other than default', () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal
+				closeDialog={ closeDialog }
+				coupon={ { duration: COUPON_DURATION_3_MONTHS } }
+			/>
+		);
+		expect( screen.getByRole( 'checkbox', { name: 'Duration' } ) ).toBeChecked();
+		expect( screen.getByRole( 'combobox', { name: '' } ) ).not.toBeDisabled();
+	} );
+
+	test( 'should show select dropdown as disabled unless toggled', () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal closeDialog={ closeDialog } coupon={ {} } />
+		);
+		expect( screen.getByRole( 'checkbox', { name: 'Duration' } ) ).not.toBeChecked();
+		expect( screen.getByRole( 'combobox', { name: '' } ) ).toBeDisabled();
+		act( () => {
+			screen.getByRole( 'checkbox', { name: 'Duration' } ).click();
+		} );
+		expect( screen.getByRole( 'checkbox', { name: 'Duration' } ) ).toBeChecked();
+		expect( screen.getByRole( 'combobox', { name: '' } ) ).not.toBeDisabled();
+	} );
+
+	test( 'should enable emails textarea if email allow list other than default', () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		const labelText = 'Limit coupon to specific emails';
+		const email_allow_list = [ '*@*.edu', '*@automattic.com' ];
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal
+				closeDialog={ closeDialog }
+				coupon={ { email_allow_list } }
+			/>
+		);
+		expect( screen.getByRole( 'checkbox', { name: labelText } ) ).toBeChecked();
+		expect( screen.getByDisplayValue( email_allow_list.join( ', ' ) ) ).not.toBeDisabled();
+	} );
+
+	test( 'should show email textarea as disabled unless toggled', () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		const labelText = 'Limit coupon to specific emails';
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal closeDialog={ closeDialog } coupon={ {} } />
+		);
+		expect( screen.getByRole( 'checkbox', { name: labelText } ) ).not.toBeChecked();
+		expect( document.getElementById( 'email_allow_list' ) ).toBeDisabled();
```

Comment:
I don't know what kind of element this is, but if it's something that can be disabled, it seems like it would be an interactive element, and there might be an accessibility issue there if it can't be queried with `getByRole()`.

Reply:
Good point, I didn't think about that. I've added `aria-label` to this so that it can be queried by `screen` without changing the design which does not have a label for this input (the label is for the toggle which enables / disables the textbox which has no label in the design because it's next to the toggle which has a label already).